Patches.py

```diff
@@ -1832,24 +1832,29 @@ def calculate_traded_flags(world):
         if world.dungeon_mq['Spirit Temple']: # Patch Spirit MQ Lobby front right chest to use permanent switch flag 0x1F
             rom.write_byte(0x2b08ce4 + 13, 0x1F)
 
+    if world.settings.shuffle_wonderitems:
+        #Patch Hyrule Castle Guards to not block the way
+        rom.write_bytes(0xCD5E30, [0x00, 0x00, 0x00, 0x00]) # nop
+        rom.write_bytes(0xCD5E7C, [0x10, 0x00, 0x00, 0x03]) # b courtyard_guards_kill
```

Comment:
So under normal circumstances, the courtyard guards would block you from collecting a wonder item?

What happens if you go all the way to Zelda at night time?

Reply:
There is a wonderitem in zelda's courtyard (you shoot the window with the mario painting in it with slingshot). Once you have met Zelda, you can no longer get back to the courtyard because the guard's block the way. This would cause a potential softlocks. This hack prevents those guards from spawning so the player can get back to the courtyard.

This does not affect the night time guards. They will still be there and prevent link from passing.

Reply:
Oh, huh, I could've sworn you had the ability to go back to Zelda and she'd tell you to keep investigating Ganondorf or something.

Nothing weird happens if you talk to Zelda in this state?

Reply:
Child trade shuffle (should have) made the guards blocking the way on subsequent visits never spawn. See https://github.com/OoTRandomizer/OoT-Randomizer/blob/6fd5d5e8989b202241ab93d43f1f9acaae1b217c/ASM/src/hacks.asm#L2701-L2708. I think this is redundant.

Reply:
yeah your hack is exactly the same as mine (you gave it to me when I first started working on wonderitem shuffle), I just wrote it in Patches.py so that it could be hacked conditionally based on if the setting was enabled. 

Would you guys prefer to keep it in hacks.asm and always just remove the guards, or add additional conditions to the hack in Patches.py?

Reply:
It's already unconditional in Dev right now, why would this PR make it conditional?

Reply:
Why am I being asked to describe the behavior of a hack that has existed unconditionally in Dev? Clearly some folks aren't even aware that the hack already exists. You guys bitch about tiny vanilla changes like this all the time, so I figured I'd ask since I have go change ~4 lines of code one way or another.

Reply:
I am not asking you to describe the behavior. I'm asking you what the reasoning behind making it conditional would be.

Reply:
My vote is keep it always on. If it's going to be different from vanilla I like it staying consistent.

I scrubbed discord questions I asked about trade shuffle and didn't see any discussion on the topic. I did have it as a bullet in the giant trade shuffle description, and no one called it out. You can hit me if people get pissed at you instead :)

Reply:
I didn't say you were the one asking about the behavior, that's why this thread started.

The "why" is because "you guys bitch about tiny vanilla changes like this all the time" and clearly didn't know this change even existed.

Reply:
In that case, my vote is also to keep it unconditional.

Reply:
Was that so fucking hard

Reply:
I really don't get why you seem so annoyed.

Reply:
Because I simply asked for your preference and get questioned about it like I'm trying to do something horrific. That's annoying.

Reply:
I could have just said to keep it unconditional because there's no reason for it to be conditional, but the fact that I didn't see any reason whatsoever suggested that I might be missing something so I asked for the reason first. I didn't mean to imply any value judgement.

Reply:
Deleted this hack from Patches.py